tests/server_encode.rs

```diff
@@ -0,0 +1,159 @@
+mod server_encode {
+    use async_h1::server::Encoder;
+    use async_std::io::Cursor;
+    use async_std::io::ReadExt;
+    use http_types::Body;
+    use http_types::Result;
+    use http_types::StatusCode;
+    use http_types::{Method, Response};
+    use pretty_assertions::assert_eq;
+
+    async fn encode_to_string(
+        response: Response,
+        len: usize,
+        method: Method,
+    ) -> http_types::Result<String> {
+        let mut buf = vec![];
+        let mut encoder = Encoder::new(response, method);
+        loop {
+            let mut inner_buf = vec![0; len];
+            let bytes = encoder.read(&mut inner_buf).await?;
+            buf.extend_from_slice(&inner_buf[..bytes]);
+            if bytes == 0 {
+                return Ok(String::from_utf8(buf)?);
+            }
+        }
+    }
+
+    async fn assert_encoded(len: usize, method: Method, response: Response, s: &str) {
+        assert_eq!(
+            encode_to_string(response, len, method)
+                .await
+                .unwrap()
+                .split("\r\n")
+                .map(|line| {
+                    if line.starts_with("date:") {
+                        "date: {DATE}"
+                    } else {
+                        line
+                    }
+                })
+                .collect::<Vec<_>>()
+                .join("\n"),
+            s
+        );
+    }
+
+    #[async_std::test]
+    async fn basic() -> Result<()> {
+        let mut res = Response::new(StatusCode::Ok);
+        res.set_body("");
```

Comment:
is this necessary?

Reply:
nope, i think it was cargo culted from a previous test. will remove